tests/test_vertex_executor.py

```diff
@@ -39,60 +40,73 @@ def add(a: int, b: int) -> int:
-    @patch(STATUS_METHOD, return_value=CANCELLED_STATUS)
-    @patch(VERTEX_PROPERTY, return_value="dummy_api")
-    def test_run(self, stage_mock, start_mock, status_mock, _):
-        """
-        Tests that the VertexExecutor.run() method calls the correct private methods
-        """
-        # swap the fs for a local fs
-        self.vertex_executor.fs = LocalFileSystem()
-        self.vertex_executor.storage_location = (
-            f"{os.path.expanduser('~')}/cascade-storage/"
+@pytest.fixture
+def vertex_executor_fixture():
+    with patch(STAGE_METHOD) as stage_mock, \
+         patch(START_METHOD, return_value="test_job") as start_mock, \
+         patch(STATUS_METHOD, return_value=CANCELLED_STATUS) as status_mock, \
+         patch(VERTEX_PROPERTY, return_value="dummy_api"), \
+         patch(FILESYSTEM, LocalFileSystem):
+        
+        vertex_executor = VertexExecutor(
+            resource=gcp_resource,
+            func=wrapped_partial(add, 1, 2),
+        )
+        vertex_executor.storage_location = (
+            f"{os.path.expanduser('~')}/cascade-storage"
         )
 
-        with self.assertRaises(VertexCancelledError):
-            self.vertex_executor.run()
-
-        stage_mock.assert_called_once()
-        start_mock.assert_called_once()
-        status_mock.assert_called_once()
-
-    def test_create_job(self):
-        """
-        Tests that a can be VertexJob is from a VertexExecutor.
-        """
-
-        test_job = self.vertex_executor.create_job()
-        assert isinstance(test_job, VertexJob)
-
-        custom_job = test_job.create_payload()
-        assert isinstance(custom_job, dict)
-
-    @patch(STORAGE_PATH, new_callable=PropertyMock, return_value="/tmp/test")
-    def test_stage(self, mock_storage_path):
-        """
-        Tests that the VertexExecutor._stage() correctly stages a function
-        """
-
+        stage_mock.reset_mock()
+        start_mock.reset_mock()
+        status_mock.reset_mock()
+
+        yield vertex_executor, stage_mock, start_mock, status_mock
+
+def test_run(vertex_executor_fixture):
+    """
+    Tests that the VertexExecutor.run() method calls the correct private methods
+    """
+    # swap the fs for a local fs
+    vertex_executor, stage_mock, start_mock, status_mock = vertex_executor_fixture
+
+    with pytest.raises(VertexCancelledError):
+        vertex_executor.run()
+
+    start_mock.assert_called_once()
+    status_mock.assert_called_once()
+    stage_mock.assert_called_once()
+
+def test_create_job(vertex_executor_fixture):
+    """
+    Tests that a VertexJob can be created from a VertexExecutor.
+    """
+    vertex_executor = vertex_executor_fixture[0]
+    test_job = vertex_executor.create_job()
+    assert isinstance(test_job, VertexJob)
+
+    custom_job = test_job.create_payload()
+    assert isinstance(custom_job, dict)
+
+@patch(VERTEX_PROPERTY, return_value="dummy_api")
+@patch(FILESYSTEM, LocalFileSystem)
+def test_stage(tmp_path):
```

Comment:
Does this need two arguments to the function since there are two patches ?

Reply:
Ah, the decorators should be removed actually since they are set up in the context manager in the function body.